pkg/util/gitutil/git_test.go

```diff
@@ -204,3 +203,60 @@ func createTestRepo(e *ptesting.Environment) {
 	e.RunCommand("git", "branch", "my/content")
 	e.RunCommand("git", "tag", "my")
 }
+
+func TestTryGetVCSInfoFromSSHRemote(t *testing.T) {
+	gitTests := []struct {
+		Remote      string
+		WantVCSInfo *VCSInfo
+	}{
+		// SSH remotes
+		{
+			"git@gitlab.com:owner-name/repo-name.git",
+			&VCSInfo{Owner: "owner-name", Repo: "repo-name", Kind: GitLabHostName},
+		},
+		{
+			"git@github.com:owner-name/repo-name.git",
+			&VCSInfo{Owner: "owner-name", Repo: "repo-name", Kind: GitHubHostName},
+		},
+		{
+			"git@bitbucket.org:owner-name/repo-name.git",
+			&VCSInfo{Owner: "owner-name", Repo: "repo-name", Kind: BitbucketHostName},
+		},
+		{
+			"git@ssh.dev.azure.com:v3/owner-name/project/repo-name.git",
+			&VCSInfo{Owner: "owner-name", Repo: "project/repo-name", Kind: AzureDevOpsHostName},
+		},
+
+		//HTTPS remotes
+		{
+			"https://gitlab-ci-token:dummytoken@gitlab.com/owner-name/repo-name.git",
+			&VCSInfo{Owner: "owner-name", Repo: "repo-name", Kind: GitLabHostName},
+		},
+		{
+			"https://github.com/owner-name/repo-name.git",
+			&VCSInfo{Owner: "owner-name", Repo: "repo-name", Kind: GitHubHostName},
+		},
+		{
+			"https://ploke@bitbucket.org/owner-name/repo-name.git",
+			&VCSInfo{Owner: "owner-name", Repo: "repo-name", Kind: BitbucketHostName},
+		},
+		{
+			"https://user@dev.azure.com/owner-name/project/_git/repo-name",
+			&VCSInfo{Owner: "owner-name", Repo: "project/_git/repo-name", Kind: AzureDevOpsHostName},
+		},
+
+		//Unknown or bad remotes
+		{"", nil},
+		{"dummy", nil},
```

Comment:
Nit: To some the this term is offensive, so consider using another placeholder like "random text" or "asdf", etc.